lib/src/mock.dart

```diff
@@ -846,7 +846,7 @@ class VerificationResult {
   void _checkTestApiMismatch() {
     try {
       Invoker.current;
-    } on CastError catch (e) {
+    } on TypeError catch (e) {
```

Comment:
Interesting that this was broken. Can you add a test that fails without this fix?

Reply:
This was not broken. It was mentioned as deprecated and the replacement was suggested. I should probably rollback this since it is not a minor tweak and should go with another PR that replaces deprecated features.

Reply:
We'll also need to ensure that our minimum SDK is at least the one that switched which error type is thrown.

Reply:
Shouldn't that be another PR where we fix the SDK versions and the deprecated types together?

Reply:
Yeah another PR would be best, I was calling it out that any PR which does make this change would also need to bump the min SDK.

Reply:
I agree. Is someone working on upgrades at the moment? If so, I would like to make a contribution

Reply:
No I don't believe anyone is working on that.